File: PowerShell for Metadata/getValueFromNestedObject.js
function getValueFromNestedObject(obj, key) {
    const keys = key.split('/');
    let value = obj;
  
    for (const k of keys) {
      if (typeof value === 'object' && value !== null && k in value) {
        value = value[k];
      } else {
        return undefined;
      }
    }
  
    return value;
  }
  
  // Example usage
  const object1 = { a: { b: { c: 'd' } } };
  const key1 = 'a/b/c';
  const value1 = getValueFromNestedObject(object1, key1);
  console.log(value1); // Output: d
  
  const object2 = { x: { y: { z: 'a' } } };
  const key2 = 'x/y/z';
  const value2 = getValueFromNestedObject(object2, key2);
  console.log(value2); // Output: a
  

File: PowerShell for Metadata/get_value_from_nested_object.py
# using some google helop i got this Pythos script. 


def get_value_from_nested_object(obj, key):
    keys = key.split('/')
    value = obj

    for k in keys:
        if isinstance(value, dict) and k in value:
            value = value[k]
        else:
            return None

    return value

# Example usage
object1 = {"a": {"b": {"c": "d"}}}
key1 = "a/b/c"
value1 = get_value_from_nested_object(object1, key1)
print(value1)  # Output: d

object2 = {"x": {"y": {"z": "a"}}}
key2 = "x/y/z"
value2 = get_value_from_nested_object(object2, key2)
print(value2)  # Output: a
